pkg/security/probe/probe_ebpf.go

```diff
@@ -1575,6 +1575,8 @@ func (p *EBPFProbe) ApplyRuleSet(rs *rules.RuleSet) (*kfilters.ApplyRuleSetRepor
 	// activity dump & security profiles
 	needRawSyscalls := p.isNeededForActivityDump(model.SyscallsEventType.String())
 
+	p.processKiller.Reset()
```

Comment:
Yes it is, the idea is that if a rule gets updated and isn't noisy anymore then we want the kill actions to be performed as usual, and if the rule is still noisy then it will be disarmed again.